File: sdks/javascript/src/resources/keywords.ts
import { AxiosInstance } from 'axios';

export interface Keyword {
  id: string;
  keyword: string;
  searchVolume?: number;
  difficulty?: number;
  cpc?: number;
  intent?: string;
  tags: string[];
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateKeywordParams {
  keyword: string;
  tags?: string[];
}

/**
 * Keywords resource
 * Manage keyword tracking
 */
export class Keywords {
  constructor(private axios: AxiosInstance) {}

  /**
   * List keywords for a project
   */
  async list(projectId: string): Promise<{ data: Keyword[] }> {
    const response = await this.axios.get(`/projects/${projectId}/keywords`);
    return response.data;
  }

  /**
   * Get a keyword by ID
   */
  async get(keywordId: string): Promise<Keyword> {
    const response = await this.axios.get(`/keywords/${keywordId}`);
    return response.data;
  }

  /**
   * Add a keyword to a project
   */
  async create(projectId: string, params: CreateKeywordParams): Promise<Keyword> {
    const response = await this.axios.post(`/projects/${projectId}/keywords`, params);
    return response.data;
  }

  /**
   * Delete a keyword
   */
  async delete(keywordId: string): Promise<void> {
    await this.axios.delete(`/keywords/${keywordId}`);
  }

  /**
   * Get keyword suggestions
   */
  async suggestions(seed: string, limit: number = 10): Promise<any[]> {
    const response = await this.axios.get('/keywords/suggestions', {
      params: { seed, limit },
    });
    return response.data;
  }
}

File: sdks/python/seo_platform/resources/keywords.py
"""Keywords resource"""
from typing import List, Dict, Any, Optional


class Keywords:
    """Keywords resource - Manage keyword tracking"""

    def __init__(self, client):
        self.client = client

    def list(self, project_id: str) -> Dict[str, Any]:
        """
        List keywords for a project

        Args:
            project_id: Project ID

        Returns:
            Dictionary with keywords data
        """
        return self.client.request('GET', f'/projects/{project_id}/keywords')

    def get(self, keyword_id: str) -> Dict[str, Any]:
        """
        Get a keyword by ID

        Args:
            keyword_id: Keyword ID

        Returns:
            Keyword data
        """
        return self.client.request('GET', f'/keywords/{keyword_id}')

    def create(
        self,
        project_id: str,
        keyword: str,
        tags: Optional[List[str]] = None,
    ) -> Dict[str, Any]:
        """
        Add a keyword to a project

        Args:
            project_id: Project ID
            keyword: Keyword text
            tags: Optional tags

        Returns:
            Created keyword data
        """
        data = {'keyword': keyword}
        if tags:
            data['tags'] = tags

        return self.client.request(
            'POST',
            f'/projects/{project_id}/keywords',
            json=data,
        )

    def delete(self, keyword_id: str) -> None:
        """
        Delete a keyword

        Args:
            keyword_id: Keyword ID
        """
        self.client.request('DELETE', f'/keywords/{keyword_id}')

    def suggestions(self, seed: str, limit: int = 10) -> List[Dict[str, Any]]:
        """
        Get keyword suggestions

        Args:
            seed: Seed keyword
            limit: Number of suggestions (default: 10)

        Returns:
            List of keyword suggestions
        """
        return self.client.request(
            'GET',
            '/keywords/suggestions',
            params={'seed': seed, 'limit': limit},
        )
